ktor-client/ktor-client-features/ktor-client-json/src/io/ktor/client/features/json/serializer/KotlinxSerializer.kt

```diff
@@ -1,39 +1,42 @@
 package io.ktor.client.features.json.serializer
 
-import io.ktor.client.call.*
-import io.ktor.client.features.json.*
-import io.ktor.client.response.*
-import io.ktor.http.*
-import io.ktor.http.content.*
-import kotlinx.serialization.json.*
-import kotlin.reflect.*
-
-typealias ReadMapper<T> = (JsonElement) -> T
-typealias WriteMapper<T> = (T) -> JsonElement
-
-class KotlinxSerializer : JsonSerializer {
-    private val readMappers = mutableMapOf<KClass<*>, ReadMapper<*>>()
-    private val writeMappers = mutableMapOf<KClass<*>, WriteMapper<*>>()
-
-    fun <T : Any> setReadMapper(type: KClass<T>, block: ReadMapper<T>) {
-        readMappers[type] = block
-    }
-
-    fun <T : Any> setWriteMapper(type: KClass<T>, block: WriteMapper<T>) {
-        writeMappers[type] = block
+import io.ktor.client.call.TypeInfo
+import io.ktor.client.features.json.JsonSerializer
+import io.ktor.client.response.HttpResponse
+import io.ktor.client.response.readText
+import io.ktor.http.ContentType
+import io.ktor.http.content.OutgoingContent
+import io.ktor.http.content.TextContent
+import kotlinx.serialization.*
+import kotlinx.serialization.json.JSON
+import kotlin.reflect.KClass
+
+expect fun serializerForPlatform(type: TypeInfo): KSerializer<*>
+
+class KotlinxSerializer (
+        private val json: JSON = JSON(
+                unquoted = false,
+                indented = false,
+                nonstrict = false,
+                context = SerialContext()
+        ),
+        block: SerialContext.() -> Unit = {}
+) : JsonSerializer {
+    init {
+        json.context?.apply {
+            block()
+        }
     }
 
     override fun write(data: Any): OutgoingContent {
         @Suppress("UNCHECKED_CAST")
-        val mapper = (writeMappers[data::class] as? (Any) -> JsonElement)
-            ?: error("There is no mapper for $data")
-
-        return TextContent(mapper(data).toString(), ContentType.Application.Json)
+        val clazz: KClass<Any> = data::class as KClass<Any>
+        val serializer = clazz.serializer()
```

Comment:
Unfortunately, `KClass<*>.serializer()` is not supported in the native plugin for now.
You could consult about it there https://github.com/Kotlin/kotlinx.serialization

Reply:
well it seems to work, for now, only non non-parameterized types though .. for more some typeinfo would do the trick
i did not want to do such a invasive change though